packages/rpc-core/src/rpc-methods/getBlock.ts

```diff
@@ -0,0 +1,149 @@
+                transactionDetails: 'full';
+                encoding: 'jsonParsed';
+            }>
+    ):
+        | (GetBlockApiResponseBase &
+              GetBlockApiResponseWithTransactions<ReturnType<GetTransactionApi['getTransaction']>> &
+              (TBlockRewards extends void ? Record<string, never> : GetBlockApiResponseWithRewards))
+        | null;
+    getBlock<
+        TMaxSupportedTransactionVersion extends TransactionVersion | void = void,
+        TBlockRewards extends boolean | void = void
+    >(
+        slot: Slot,
+        config: GetBlockCommonConfig<TMaxSupportedTransactionVersion, TBlockRewards> &
+            Readonly<{
+                transactionDetails: 'full';
+                encoding: 'base64';
+            }>
+    ):
+        | (GetBlockApiResponseBase &
+              GetBlockApiResponseWithTransactions<ReturnType<GetTransactionApi['getTransaction']>> &
+              (TBlockRewards extends void ? Record<string, never> : GetBlockApiResponseWithRewards))
+        | null;
+    getBlock<
+        TMaxSupportedTransactionVersion extends TransactionVersion | void = void,
+        TBlockRewards extends boolean | void = void
+    >(
+        slot: Slot,
+        config: GetBlockCommonConfig<TMaxSupportedTransactionVersion, TBlockRewards> &
+            Readonly<{
+                transactionDetails: 'full';
+                encoding: 'base58';
+            }>
+    ):
+        | (GetBlockApiResponseBase &
+              GetBlockApiResponseWithTransactions<ReturnType<GetTransactionApi['getTransaction']>> &
+              (TBlockRewards extends void ? Record<string, never> : GetBlockApiResponseWithRewards))
+        | null;
+    getBlock<
+        TMaxSupportedTransactionVersion extends TransactionVersion | void = void,
+        TBlockRewards extends boolean | void = void
+    >(
+        slot: Slot,
+        config?: GetBlockCommonConfig<TMaxSupportedTransactionVersion, TBlockRewards> &
+            Readonly<{
+                transactionDetails: 'full';
+                encoding?: 'json';
+            }>
+    ):
+        | (GetBlockApiResponseBase &
+              GetBlockApiResponseWithTransactions<ReturnType<GetTransactionApi['getTransaction']>> &
+              (TBlockRewards extends void ? Record<string, never> : GetBlockApiResponseWithRewards))
+        | null;
+    getBlock<
+        TMaxSupportedTransactionVersion extends TransactionVersion | void = void,
+        TBlockRewards extends boolean | void = void
+    >(
+        slot: Slot,
+        config: GetBlockCommonConfig<TMaxSupportedTransactionVersion, TBlockRewards> &
+            Readonly<{
+                transactionDetails: 'accounts';
+                encoding?: 'jsonParsed' | 'base64' | 'base58' | 'json';
+            }>
+    ):
+        | (GetBlockApiResponseBase &
+              GetBlockApiResponseWithSignatures &
+              (TBlockRewards extends void ? Record<string, never> : GetBlockApiResponseWithRewards))
+        | null;
```

Comment:
This looks the same as `transactionDetails: 'signatures'`. From the docs it says "transaction details only include signatures and an annotated list of accounts in each transaction."

Testing this:
```
curl --location 'https://explorer-api.devnet.solana.com/' \
--header 'authority: explorer-api.devnet.solana.com' \
--header 'accept-language: en-US,en;q=0.9' \
--header 'content-type: application/json' \
--header 'origin: https://explorer.solana.com' \
--header 'referer: https://explorer.solana.com/' \
--data '{
    "method": "getBlock",
    "jsonrpc": "2.0",
    "params": [
        241879235,
        {
            "commitment": "confirmed",
            "maxSupportedTransactionVersion": 0,
            "transactionDetails": "accounts"
        }
    ],
    "id": "0e5f2a86-413e-4d03-b9ba-17b4b5cb4d3e"
}'
```

It looks like the response includes (just first tx):

```json
{
  "transactions": [
    {
      "meta": {
        "err": null,
        "fee": 5000,
        "postBalances": [
          2533752548939,
          100000000000,
          1
        ],
        "postTokenBalances": [],
        "preBalances": [
          2533752553939,
          100000000000,
          1
        ],
        "preTokenBalances": [],
        "rewards": [],
        "status": {
          "Ok": null
        }
      },
      "transaction": {
        "accountKeys": [
          {
            "pubkey": "dv1ZAGvdsz5hHLwWXsVnM94hWf1pjbKVau1QVkaMJ92",
            "signer": true,
            "source": "transaction",
            "writable": true
          },
          {
            "pubkey": "5ZWgXcyqrrNpQHCme5SdC5hCeYb2o3fEJhF7Gok3bTVN",
            "signer": false,
            "source": "transaction",
            "writable": true
          },
          {
            "pubkey": "Vote111111111111111111111111111111111111111",
            "signer": false,
            "source": "transaction",
            "writable": false
          }
        ],
        "signatures": [
          "HweSFWxvdWWADJQ6MMCkUJYscCdkLYBteFZmN8PHkHmp7pN2bzWw2JvGcgdZKPbSvGhLrX5svMSqtgw62nssxsk"
        ]
      },
      "version": "legacy"
    },
    ... // other transactions
}
```

So it's a partial version of meta (docs say "Transaction metadata is limited to only: fee, err, pre_balances, post_balances, pre_token_balances, and post_token_balances.") and then just `accountKeys`

It seems to completely ignore encoding and always use the `jsonParsed` style for `accountKeys`